src/codewhisperer/util/runtimeLanguageContext.ts

```diff
@@ -3,103 +3,147 @@
      * Value: CodeWhispererLanguageId
      */
-    private supportedLanguageMap: ConstantMap<CodeWhispererConstants.SupportedLanguage, CodewhispererLanguage>
+    private supportedLanguageMap: ConstantMap<
+        CodeWhispererConstants.PlatformLanguageId | CodewhispererLanguage,
+        CodewhispererLanguage
+    >
 
     /**
      * A map storing CodeWhisperer supported programming language with key: vscLanguageId and value: language extension
      * Key: vscLanguageId
      * Value: language extension
      */
-    private supportedLanguageExtensionMap: ConstantMap<CodeWhispererConstants.SupportedLanguage, string>
-
-    // A set contains vscode languageId and CodeWhispererLanguage
-    private supportedLanguageSet = new Set<string>()
+    private supportedLanguageExtensionMap: ConstantMap<CodewhispererLanguage, string>
 
     constructor() {
-        this.supportedLanguageMap = createConstantMap<CodeWhispererConstants.SupportedLanguage, CodewhispererLanguage>({
-            java: 'java',
-            python: 'python',
-            javascriptreact: 'jsx',
-            javascript: 'javascript',
-            typescript: 'typescript',
-            typescriptreact: 'tsx',
-            csharp: 'csharp',
+        this.supportedLanguageMap = createConstantMap<
+            CodeWhispererConstants.PlatformLanguageId | CodewhispererLanguage,
+            CodewhispererLanguage
+        >({
             c: 'c',
-            c_cpp: 'cpp',
             cpp: 'cpp',
+            csharp: 'csharp',
+            c_cpp: 'cpp',
             go: 'go',
+            java: 'java',
+            javascript: 'javascript',
+            javascriptreact: 'jsx',
+            jsx: 'jsx',
             kotlin: 'kotlin',
+            plaintext: 'plaintext',
             php: 'php',
+            python: 'python',
             ruby: 'ruby',
             rust: 'rust',
             scala: 'scala',
             sh: 'shell',
+            shell: 'shell',
             shellscript: 'shell',
             sql: 'sql',
-        })
-        this.supportedLanguageExtensionMap = createConstantMap<CodeWhispererConstants.SupportedLanguage, string>({
-            java: 'java',
-            python: 'py',
-            javascriptreact: 'jsx',
-            javascript: 'js',
-            typescript: 'ts',
+            tsx: 'tsx',
+            typescript: 'typescript',
             typescriptreact: 'tsx',
-            csharp: 'cs',
+        })
+        this.supportedLanguageExtensionMap = createConstantMap<CodewhispererLanguage, string>({
             c: 'c',
-            c_cpp: 'cpp',
             cpp: 'cpp',
+            csharp: 'cs',
             go: 'go',
+            java: 'java',
+            javascript: 'js',
+            jsx: 'jsx',
             kotlin: 'kt',
+            plaintext: 'txt',
             php: 'php',
+            python: 'py',
             ruby: 'rb',
             rust: 'rs',
             scala: 'scala',
-            sh: 'sh',
-            shellscript: 'sh',
```

Comment:
is there a reason to remove `sh: 'sh'`? IIRC is this added to support a case in C9? Same applies to the above `c_cpp: 'cpp'`

Reply:
here, normalize func is being called https://github.com/aws/aws-toolkit-vscode/pull/3940/files#diff-0509671083ef22655748504f51cc4ad12de62ff0ec7a4f317f57d6bf24652838R138

Reply:
removing from this map is to avoid ppl have to manually add so many different [languageId, extension] pair in this map. So I move the whole thing to `this.supportedLanguageMap` at L50.

Reply:
also here `isLanguageSupported()` https://github.com/aws/aws-toolkit-vscode/pull/3940/files#diff-0509671083ef22655748504f51cc4ad12de62ff0ec7a4f317f57d6bf24652838R179, will normalize languageId first then check

Reply:
basically every methods in this class will normalize the provided language id first then do its work

Reply:
yes a functional test would be nice, I will add this and test it on C9

Reply:
ah I see.

Reply:
How can we prevent people from not using `normalize` first in the future?

Reply:
basically "normalized" language is of type `CodewhispererLanguage` so we can change all functions which take language as one of their argument to be of type `CodewhispererLanguage`

Reply:
so that we can have compile time check

Reply:
which I think it apply to most of our functions but of course something like `getTracker()` is not this case I can update this in this PR as well